src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserManagementService.cs

```diff
@@ -54,6 +62,75 @@ public async Task<List<UserViewModel>> ListAsync()
+        public async Task<UserReleaseRoleViewModel> GetUserReleaseRole(Guid userId, Guid userReleaseRoleId)
+        {
+            return await _contentDbContext.UserReleaseRoles
+                .Where(x => x.Id == userReleaseRoleId && x.UserId == userId)
+                .Select(x => new UserReleaseRoleViewModel
+                {
+                    Id = x.Id,
+                    Publication = _contentDbContext.Publications
+                        .Where(p => p.Releases.Any(r => r.Id == x.ReleaseId))
+                        .Select(p => new IdTitlePair {Id = p.Id, Title = p.Title}).FirstOrDefault(),
+                    Release = _contentDbContext.Releases
+                        .Where(r => r.Id == x.ReleaseId)
+                        .Select(r => new IdTitlePair {Id = r.Id, Title = r.Title}).FirstOrDefault(),
+                    ReleaseRole = new EnumExtensions.EnumValue {Name = x.Role.GetEnumLabel(), Value = 0}
+                }).FirstOrDefaultAsync();
+        }
+
+        public async Task<Either<ActionResult, UserReleaseRoleViewModel>> AddUserReleaseRole(Guid userId,
+            UserReleaseRoleSubmission userReleaseRole)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<Release>(userReleaseRole.ReleaseId)
+                .OnSuccess(_ => CheckIfUserAlreadyHasReleaseRole(userId, userReleaseRole))
+                .OnSuccess(async _ =>
+                {
+                    var newReleaseRole = new UserReleaseRole
+                    {
+                        ReleaseId = userReleaseRole.ReleaseId,
+                        Role = userReleaseRole.ReleaseRole,
+                        UserId = userId
+                    };
+
+                    _contentDbContext.Add(newReleaseRole);
+
+                    await _contentDbContext.SaveChangesAsync();
+
+                    var response = await GetUserReleaseRole(userId, newReleaseRole.Id);
+
+                    SendNewReleaseRoleEmail(userId, response.Publication, response.Release, response.ReleaseRole);
+
+                    return response;
+                });
+        }
+
+
+        public async Task<Either<ActionResult, bool>> RemoveUserReleaseRole(Guid userId, Guid userReleaseRoleId)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<UserReleaseRole>(userReleaseRoleId)
+                .OnSuccessDo(async () =>
+                {
+                    var entityToRemove =
+                        await _contentDbContext.UserReleaseRoles.FirstOrDefaultAsync(r =>
+                            r.Id == userReleaseRoleId && r.UserId == userId);
+                    _contentDbContext.Remove(entityToRemove);
+                    await _contentDbContext.SaveChangesAsync();
+                })
+                .OnSuccess(_ => true);
+        }
+
+        public async Task<List<IdTitlePair>> ListReleasesAsync()
+        {
+            return await _contentDbContext.Releases.Select(r => new IdTitlePair
```

Comment:
I think you should `.Include` publication here. This wouldn't work called on its own unless something else has already loaded the Publication into the Context.

Reply:
I've not encountered that error but it probably wouldn't hurt, i think in theory if your using select it executes and returns that specific sql command, i think it would break if you used orderby after the select on something that comes from the publication.